src/services/analysis.ts

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios'
+
+// Utils
+import { delay, getRequestHeaders } from '../utils'
+
+export const analyze = async (data: {}, accessToken: string, url: string) => {
+  const headers = getRequestHeaders(accessToken)
+
+  try {
+    const response = await axios.post(url, { data }, { headers })
+
+    const uuidUrl = `${url}/${response.data.uuid}`
+    let status = response.data.status
+
+    // Initial status check
+    if (status === 'Finished') {
+      return getIssues(accessToken, `${uuidUrl}/issues`)
+    }
+
+    for (let i = 0; i < 10; i++) {
+      await delay(3000)
+
+      status = (await getStatus(accessToken, uuidUrl)).status
+
+      if (status === 'Finished') {
+        return getIssues(accessToken, `${uuidUrl}/issues`)
+      }
+    }
+  } catch (err) {
+    throw new Error(err.response.data.error)
+  }
+}
+
+export const getIssues = (accessToken: string, url: string) => {
+  const headers = getRequestHeaders(accessToken)
+
+  return axios
```

Comment:
Can we use `async/await` here?

Reply:
There's no need to await on the promise here. We can directly return the promise and the function which is calling it should take care of handling the promise